sdks/java/io/elasticsearch-tests/elasticsearch-tests-common/src/test/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIOTestCommon.java

```diff
@@ -547,6 +550,42 @@ void testWritePartialUpdate() throws Exception {
     assertEquals(numDocs / 2, countByMatch(connectionConfiguration, restClient, "group", "1"));
   }
 
+  /** Tests partial updates with errors by adding some invalid info to test set. */
+  void testWritePartialUpdateWithErrors() throws Exception {
+    // put a mapping to simulate error of insertion
+    ElasticSearchIOTestUtils.setIndexMapping(
+        restClient, connectionConfiguration.getIndex(), connectionConfiguration.getType());
+
+    if (!useAsITests) {
+      ElasticSearchIOTestUtils.insertTestDocuments(connectionConfiguration, numDocs, restClient);
+    }
+
+    // try to partial update a document with an incompatible date format for the age to generate
+    // an update error
```

Comment:
Nice workaround to the flakey test issue

Reply:
More over, the shard overloading with incorrect documents had impacts on the whole cluster stability leading to flakiness in other tests. By having only one error doc updated, we avoid shard overloading.

Reply:
Bonus